### PR TITLE
[Gmp]  add cflags to build commands for compiletime tools

### PR DESCRIPTION
**Pull request by xandox**
add `cflags` to build commands for compile time tools. Additional PR for #15605

- #### Which triplets are supported/not supported? same
- #### Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  Yes

### CODE REVIEW

**Comment by JackBoosY** [comment]
```
CMake Error at D:/installed/x64-windows-static/share/yasm-tool-helper/yasm-tool-helper.cmake:16 (message):
  Cross-targetting and x64 ports requiring yasm require the x86-windows
  yasm-tool to be available.  Please install yasm-tool:x86-windows first.
Call Stack (most recent call first):
  ports/gmp/portfile.cmake:19 (yasm_tool_helper)
  scripts/ports.cmake:142 (include)
```
You should add yasm-tool as a host dependency.

File: ports/gmp/vcpkg.json [hunk 1]
```diff
@@ -1,17 +1,24 @@
 {
   "name": "gmp",
   "version-string": "6.2.1",
-  "port-version": 4,
+  "port-version": 6,
```
Comment (JackBoosY) [inline ports/gmp/vcpkg.json:4]:
```suggestion
  "port-version": 5,
```

File: versions/baseline.json [hunk 2]
```diff
@@ -2302,7 +2302,7 @@
     },
     "gmp": {
       "baseline": "6.2.1",
-      "port-version": 4
+      "port-version": 6
```
Comment (JackBoosY) [inline versions/baseline.json:2305]:
```suggestion
      "port-version": 5
```

File: versions/g-/gmp.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "20ee3a950a47cf9573f15825114b835fc2e85ac3",
+      "version-string": "6.2.1",
+      "port-version": 6
+    },
```
Comment (JackBoosY) [inline versions/g-/gmp.json:7]:
```suggestion
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by xandox** [comment]
@JackBoosY looks like there are some problems with versions.

File: versions/g-/gmp.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "4c7fd1c1653a634159b15a5fba32592f49554a2a",
```
Comment (JackBoosY) [inline versions/g-/gmp.json:4]:
```suggestion
      "git-tree": "90eb02f55e96530420b8e83209651a9b91b7b002",
```

**Comment by strega-nil-ms** [comment]
LGTM, thanks @xandox :)

**Comment by acgetchell** [comment]
I think this broke cgal:x64-windows:

https://github.com/microsoft/vcpkg/issues/17584

**Comment by acgetchell** [comment]
Confirmed, it does.

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/history

**Pull request merged**